graphql-database-manager-dynamo/src/main/java/com/fleetpin/graphql/database/manager/dynamo/DynamoDb.java

```diff
@@ -258,6 +267,220 @@ public int maxBatchSize() {
+	
+				builder
+				.keyConditionExpression("organisationIdType = :organisationIdType AND startsWithUpdatedAt BETWEEN :fromId  AND :toId")
+	            .expressionAttributeValues(keyConditions).indexName("startsWithUpdatedAt").filterExpression("updatedAt BETWEEN :fromUpdatedAt AND :toUpdatedAt AND begins_with (id, :idStarts)");
+    		} else if (queryHistory.getFromUpdatedAt() != null ) {
+    			var fromIdAttribute = toUpdatedAtId(starts, queryHistory.getFromUpdatedAt().toEpochMilli(), true);
+                var toIdAttribute = toUpdatedAtId(starts, Long.MAX_VALUE, false);
+    	        Map<String, AttributeValue> keyConditions = new HashMap<>();
+    	        keyConditions.put(":fromId", fromIdAttribute);
+    	        keyConditions.put(":toId", toIdAttribute);
+    	        keyConditions.put(":organisationIdType", organisationIdType);
+    	        
+    	        var fromUpdatedAt = AttributeValue.builder().n(Long.toString(queryHistory.getFromUpdatedAt().toEpochMilli())).build();
+    	        keyConditions.put(":idStarts", idStarts);
+    	        keyConditions.put(":fromUpdatedAt", fromUpdatedAt);
+
+    			builder
+    			.keyConditionExpression("organisationIdType = :organisationIdType AND startsWithUpdatedAt BETWEEN :fromId  AND :toId")
+                .expressionAttributeValues(keyConditions).indexName("startsWithUpdatedAt").filterExpression("updatedAt >= :fromUpdatedAt AND begins_with (id, :idStarts)");
+    		} else if (queryHistory.getToUpdatedAt() != null ) {
+    			var fromIdAttribute = toUpdatedAtId(starts, 0L, true);
+                var toIdAttribute = toUpdatedAtId(starts, queryHistory.getToUpdatedAt().toEpochMilli(), false);
+    	        Map<String, AttributeValue> keyConditions = new HashMap<>();
+    	        keyConditions.put(":fromId", fromIdAttribute);
+    	        keyConditions.put(":toId", toIdAttribute);
+    	        keyConditions.put(":organisationIdType", organisationIdType);
+    	        
+    	        var toUpdatedAt = AttributeValue.builder().n(Long.toString(queryHistory.getToUpdatedAt().toEpochMilli())).build();
+    	        keyConditions.put(":idStarts", idStarts);
+    	        keyConditions.put(":toUpdatedAt", toUpdatedAt);
+
+    			builder
+    			.keyConditionExpression("organisationIdType = :organisationIdType AND startsWithUpdatedAt BETWEEN :fromId  AND :toId")
+                .expressionAttributeValues(keyConditions).indexName("startsWithUpdatedAt").filterExpression("updatedAt <= :toUpdatedAt AND begins_with (id, :idStarts)");
+    		}
+    	}
+        
+        var toReturn = new ArrayList<T>();
+        return client.queryPaginator(builder.build())
+        .subscribe(response -> {
+            response.items().forEach(item -> toReturn.add(new DynamoItem(historyTable, item).convertTo(mapper, queryHistory.getType())));
+        }).thenApply(__ -> {
+            return toReturn;
+        });
+    }
+    
+	private AttributeValue toId(String id) {
```

Comment:
would move toId toRevisionId toUpdatedAtId to a util class I think.

Reply:
I would have agree here :)